acceptance-test/src/acceptance-test/java/org/zalando/nakadi/webservice/StorageControllerAT.java

```diff
@@ -21,18 +27,32 @@ public void shouldChangeDefaultStorageWhenRequested() throws Exception {
                         "\"addresses\":[{\"address\":\"zookeeper\", \"port\":2181}]" +
                         "}" +
                         "},\"storage_type\": \"kafka\"}")
-                .contentType(JSON).post("/storages");
+                .contentType(JSON).post("/storages")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED);
 
         NakadiTestUtils.createEventTypeInNakadi(EventTypeTestBuilder.builder().name("event_a").build());
         String storageId = (String) NakadiTestUtils.listTimelines("event_a").get(0).get("storage_id");
         Assert.assertEquals("default", storageId);
 
-        given().contentType(JSON).put("/storages/default/default-test");
-        NakadiTestUtils.createEventTypeInNakadi(EventTypeTestBuilder.builder().name("event_b").build());
-        storageId = (String) NakadiTestUtils.listTimelines("event_b").get(0).get("storage_id");
-        Assert.assertEquals("default-test", storageId);
+        given().contentType(JSON).put("/storages/default/default-test")
+                .then()
+                .statusCode(HttpStatus.SC_OK);
 
-        // cleanup
+        // change of default storage is eventually consistent and relies on zookeeper watcher being processed
+        waitFor(() -> {
```

Comment:
I think this could wait forever?

Reply:
@lmontrieux it has a default value. I'm not sure if I have to annotate the test though, to make sure it waits at least for this amount of time `waitFor(runnable, 10000L, 500);`.

Reply:
Since it has a default value I guess it should be fine

Reply:
I imagine the value is at least a few seconds